File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  message = message.trim()

  if (message === '') return "Fine. Be that way!"

  const isQuestion = message.endsWith('?')
  const isYelled = message === message.toUpperCase() && /\p{Lu}/u.test(message)

  if (isQuestion && isYelled)
    return "Calm down, I know what I'm doing!"
  else if (isQuestion)
    return "Sure."
  else if (isYelled)
    return "Whoa, chill out!"
  else
    return "Whatever."
}

File: solutions/python/bob/1/bob.py
def response(hey_bob: str) -> str:
    sentence = hey_bob.strip()

    if sentence == "":
        return "Fine. Be that way!"

    is_question = sentence.endswith("?")
    is_yelled = sentence == sentence.upper() and any(c.isalpha() for c in sentence)

    match (is_question, is_yelled):
        case (True, True):
            return "Calm down, I know what I'm doing!"
        case (True, False):
            return "Sure."
        case (False, True):
            return "Whoa, chill out!"
        case _:
            return "Whatever."
